extensions/firmware/org.eclipse.smarthome.firmware.filesystem.test/src/test/groovy/org/eclipse/smarthome/firmware/filesystem/FileSystemFirmwareProviderTest.groovy

```diff
@@ -0,0 +1,724 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.firmware.filesystem
+
+import static org.hamcrest.CoreMatchers.*
+import static org.junit.Assert.*
+import static org.junit.matchers.JUnitMatchers.*
+
+import java.nio.file.Files;
+
+import org.eclipse.smarthome.config.core.ConfigConstants
+import org.eclipse.smarthome.core.thing.ThingTypeUID
+import org.eclipse.smarthome.core.thing.firmware.Firmware
+import org.eclipse.smarthome.core.thing.firmware.FirmwareUID
+import org.junit.After
+import org.junit.AfterClass
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Ignore;
+import org.junit.Test
+
+/**
+ * Testing the {@link FileSystemFirmwareProvider}.
+ *
+ * @author Andre Fuechsel
+ * @author Dimitar Ivanov
+ */
+class FileSystemFirmwareProviderTest {

```

Comment:
You are missing the run configuration for this test. This will make it easily runnable in Eclipse.